server/registry/actions_artifacts.go

```diff
@@ -60,6 +61,9 @@ func (s *RegistryServer) CreateArtifact(ctx context.Context, req *rpc.CreateArti
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
+	if system.Reserved(parent.Artifact(req.ArtifactId)) {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
```

Comment:
This needs to create or use a different error than the nil one left over from the parseArtifactParent() call above it.

Reply:
oops, yes, I would have caught that if I'd written a test :)

Reply:
Fixed and updated with tests, thanks!

(I wrote the tests before fixing the code and the code crashed on the err.Error() call)